sycl/CMakeLists.txt

```diff
@@ -220,20 +220,28 @@ if (MSVC)
   list(APPEND SYCL_RT_LIBS sycld)
 endif()
 
+function(add_libcxx_compile_options LIB_NAME)
+  target_compile_options(${LIB_NAME} PRIVATE -nostdinc++)
+  target_include_directories(${LIB_NAME} PRIVATE "${SYCL_LIBCXX_INCLUDE_PATH}")
+endfunction(add_libcxx_include_option)
+
+function(add_libcxx_link_options LIB_NAME)
+  target_link_libraries(${LIB_NAME} PRIVATE "-L${SYCL_LIBCXX_LIBRARY_PATH}" -Wl,-rpath,${SYCL_LIBCXX_LIBRARY_PATH} -nodefaultlibs -lc++ -lc++abi -lm -lc -lgcc_s -lgcc)
+endfunction(add_libcxx_link_options)
+
```

Comment:
I think, this file becomes quite a mess. It's hard to read and navigate here. How about refactoring this to something like `cmake/modules/SYCLUtils.cmake`? Not a very strong suggestion, though.

@bader, @vladimirlaz  what do you think?

Reply:
Hi, @alexbatashev 
I think the refactoring work can be done in a separate PR, is it OK?
Thanks very much.

Reply:
Yes, that's right